src/jdk.compiler/share/classes/com/sun/tools/javac/util/List.java

```diff
@@ -419,6 +419,9 @@ public A last() {
 
     @SuppressWarnings("unchecked")
     public <Z> List<Z> map(Function<A, Z> mapper) {
+        if (isEmpty()) {
+            return (List<Z>)this;
```

Comment:
Use `List.nil()` instead, in case this list gets mutated later:
```suggestion
            return List.<Z>nil();
```

Reply:
Wouldn't this change the logic compared to what it did before? If the list gets mutated later - e.g. via `setTail` - this would fail with `List.nil()`.
Do I miss something here?

Reply:
I meant the list on which `map(…)` is called.

Reply:
Same "problem". This changes behaviour compared to what it does now. I don't have anything against that per se, but an optimization imho shouldn't really change the behaviour.

Reply:
I agree with @dreis2211 here. Someone might want to use/mutate the returned `List` later on.

Reply:
This won’t be a change in behaviour as `ListBuffer::toList()` returns `List.nil()` when called on an empty `ListBuffer`.

Reply:
The behavior before returns the original list because `changed` is false.

Reply:
Jumping in here... I think what @ExE-Boss is getting at is that every empty `List` is the same object, i.e., `List.EMPTY_LIST`, i.e., `List.nil()`. In other words, it's not possible to ever have an empty `List` that's **not** `List.nil()`.

So while it appears that replacing `(List<Z>)this` with `List.<Z>nil()` is a change in behavior, in the special case that `this.isEmpty()` is true (which is the case here), it's not a change in behavior.

Of course this is only due to a quirk in the implementation, but the statement is still correct.

Reply:
> Would you be willing to sponsor this change by any chance?

Yes except I'm a project "author" not a "committer" so can't do that yet.

Reply:
But you could already create a ticket? That would already help :)